internal/host/hostcommands/install_cmd.go

```diff
@@ -305,15 +308,22 @@ func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host, inve
 		}
 	}
 
+	var installationDisk *models.Disk
 	for _, disk := range inventory.Disks {
 		if disk.ID == host.InstallationDiskID {
-			if disk.DriveType == models.DriveTypeMultipath {
-				installerArgs = append(installerArgs, "--append-karg", "root=/dev/disk/by-label/dm-mpath-root", "--append-karg", "rw", "--append-karg", "rd.multipath=default")
-			} else if disk.DriveType == models.DriveTypeISCSI {
-				// Currently only allowed on the OCI platform
-				installerArgs = append(installerArgs, "--append-karg", "rd.iscsi.firmware=1")
-			}
+			installationDisk = disk
+			break
+		}
+	}
+
+	if installationDisk != nil && installationDisk.DriveType == models.DriveTypeMultipath {
+		installerArgs = append(installerArgs, "--append-karg", "root=/dev/disk/by-label/dm-mpath-root", "--append-karg", "rw", "--append-karg", "rd.multipath=default")
+	} else if installationDisk != nil && installationDisk.DriveType == models.DriveTypeISCSI {
+		installerArgs, err = appendISCSIArgs(installerArgs, installationDisk, inventory, hasUserConfiguredIP)
+		if err != nil {
+			return "", err
 		}
+
 	}
```

Comment:
first check only once if the disk is nil - no need to go throw another if 

```suggestion
	if installationDisk != nil {
		if installationDisk.DriveType == models.DriveTypeMultipath {
			installerArgs = append(installerArgs, "--append-karg", "root=/dev/disk/by-label/dm-mpath-root", "--append-karg", "rw", "--append-karg", "rd.multipath=default")
		} else if  installationDisk.DriveType == models.DriveTypeISCSI {
			installerArgs, err = appendISCSIArgs(installerArgs, installationDisk, inventory, hasUserConfiguredIP)
			if err != nil {
				return "", err
			}
		}
	}
```

Reply:
will propably refactor to avoid too many levels of ifs